state-sync/state-sync-v2/data-streaming-service/src/streaming_client.rs

```diff
@@ -0,0 +1,343 @@
+/// state sync) can process data notifications in the order they're received.
+/// For example, if we're streaming transactions with proofs, state sync can
+/// assume the transactions are returned in monotonically increasing versions.
+#[async_trait]
+pub trait DataStreamingClient {
+    /// Fetches all account states at the specified version. The specified
+    /// version must be an epoch ending version, otherwise an error will be
+    /// returned. Account state proofs are at the same specified version.
+    async fn get_all_accounts(&self, version: Version) -> Result<DataStreamListener, Error>;
+
+    /// Fetches all epoch ending ledger infos starting at `start_epoch`
+    /// (inclusive) and ending at the last known epoch advertised in the network.
+    async fn get_all_epoch_ending_ledger_infos(
+        &self,
+        start_epoch: Epoch,
+    ) -> Result<DataStreamListener, Error>;
+
+    /// Fetches all transactions with proofs from `start_version` to
+    /// `end_version` (inclusive), where the proof versions can be up to the
+    /// specified `max_proof_version` (inclusive). If `include_events` is true,
+    /// events are also included in the proofs.
+    async fn get_all_transactions(
+        &self,
+        start_version: Version,
+        end_version: Version,
+        max_proof_version: Version,
+        include_events: bool,
+    ) -> Result<DataStreamListener, Error>;
+
+    /// Fetches all transaction outputs with proofs from `start_version` to
+    /// `end_version` (inclusive), where the proof versions can be up to the
+    /// specified `max_proof_version` (inclusive).
+    async fn get_all_transaction_outputs(
+        &self,
+        start_version: Version,
+        end_version: Version,
+        max_proof_version: Version,
+    ) -> Result<DataStreamListener, Error>;
+
+    /// Refetches the payload for the data notification corresponding to the
+    /// specified `notification_id`.
+    ///
+    /// Note: this is required because data payloads may be invalid, e.g., due
+    /// to invalid or malformed data returned by a misbehaving peer or a failure
+    /// to verify a proof. The refetch request forces a refetch of the payload
+    /// and the `refetch_reason` notifies the streaming service as to why the
+    /// payload must be refetched.
+    async fn refetch_notification_payload(
+        &self,
+        notification_id: NotificationId,
+        refetch_reason: PayloadRefetchReason,
+    ) -> Result<DataStreamListener, Error>;
+
+    /// Continuously streams transactions with proofs as the blockchain grows.
+    /// The stream starts at `start_version` and `start_epoch` (inclusive).
+    /// Transaction proof versions are tied to ledger infos within the same
+    /// epoch, otherwise epoch ending ledger infos will signify epoch changes.
+    /// If `include_events` is true, events are also included in the proofs.
+    async fn continuously_stream_transactions(
```

Comment:
Nit: Personally, just saying `stream_transactions` would mean continuously to me, but if it's more clear with `continuously` to you then we can leave it.

Reply:
Yeah, I think it's clearer, especially since everything returns a stream. `Continuously` in my mind tells me, there's no end. Naming is hard :P